packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -713,3 +713,21 @@ class UiKitViewController {
     await SystemChannels.platform_views.invokeMethod<void>('dispose', id);
   }
 }
+
+/// An interface for a controlling a single platform view.
+///
+/// Used by [PlatformViewSurface] to interface with the platform it embeds.
+abstract class PlatformViewController {
+
+  /// The platform view id associated with this controller.
+  ///
+  /// The id should always be unique and non-negative. And it must be set
+  /// to a non-null value at the construction, or right after construction.
+  ///
+  /// The id should be generated with [PlatformViewsRegistry.getNextPlatformViewId].
+  /// If there is an id provided to you when constructing the concrete instance
```

Comment:
I don't think it's common for our doc comments to address the reader ("to you"). Maybe just mention that this is normally the [PlatformViewCreationParams.id] passed to PlatformViewCreationCallback. (we can do that in the PR we add these).